src/components/MediaDevices/SpeakerCheck.tsx

```diff
@@ -0,0 +1,174 @@
+/// <reference types="vite/client" />
+// needed to load the .wav file
+
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import { useDevices } from '@daily-co/daily-react';
+import { Link } from 'react-router-dom';
+
+import { AudioAnalyser } from '../shared/AudioVisualiser/AudioAnalyser.ts';
+import { AudioVisualiser } from '../shared/AudioVisualiser/AudioVisualiser.tsx';
+import { Card } from '../shared/Card/Card';
+import { TroubleShooting } from '../shared/TroubleShooting/TroubleShooting';
+
+import sound from '../shared/AudioVisualiser/sound.wav';
+import { logger } from '../../utils/Logger';
+
+export const SpeakerCheck: React.FC = () => {
+  const { speakers, setSpeaker } = useDevices();
+
+  const audioRef = useRef<HTMLAudioElement>(null);
+  const [isPlaying, setIsPlaying] = useState(false);
+  const [speakerAnalyser, setSpeakerAnalyser] = useState<AudioAnalyser | null>(
+    null,
+  );
+
+  const selectedSpeaker = useMemo(() => {
+    return speakers.find((s) => s.selected);
+  }, [speakers]);
+
+  const [showTroubleshooting, setShowTroubleshooting] = useState(false);
+  const toggleTroubleShooting = () => {
+    if (showTroubleshooting) {
+      setShowTroubleshooting(false);
+    } else {
+      setShowTroubleshooting(true);
+    }
+  };
+
+  useEffect(() => {
+    const audio = audioRef.current;
+    const handlePlaying = () => {
+      setIsPlaying(true);
+    };
+    const handleEnded = () => {
+      setIsPlaying(false);
+    };
+    if (audio) {
+      audio.addEventListener('playing', handlePlaying);
+      audio.addEventListener('ended', handleEnded);
+    }
+    return () => {
+      audio?.removeEventListener('playing', handlePlaying);
+      audio?.removeEventListener('ended', handleEnded);
+    };
+  }, [audioRef]);
+
+  useEffect(() => {
+    const newAnalyser = new AudioAnalyser();
+
+    if (audioRef.current && isPlaying) {
+      newAnalyser.analyseAudioElement(audioRef.current);
+      newAnalyser.createOutput();
+      setSpeakerAnalyser(newAnalyser);
+    }
+
+    // Clean-up
+    return () => {
+      newAnalyser.disconnectAnalyser();
+    };
+  }, [audioRef, isPlaying, selectedSpeaker]);
+
+  const updateSpeakers = (ev: React.ChangeEvent<HTMLSelectElement>) => {
+    const audioEl = audioRef.current;
+    const isPaused = audioEl?.paused;
+
+    if (!isPaused) {
+      audioEl?.pause();
+    }
+    setSpeaker(ev.target.value)
+      .then(async () => {
+        speakerAnalyser?.setSinkId(ev.target.value);
+        if (!isPaused) {
+          await audioEl?.play();
+        }
+      })
+      .catch((err) => logger.error(err));
+  };
+
+  useEffect(() => {
+    if (!audioRef.current) {
+      return;
+    }
+    audioRef.current.play();
+    audioRef.current.loop = true;
```

Comment:
Can we set the `loop` attribute on the `<audio>` element directly instead of setting it here?
Also calling `play()` could be replaced with setting the `autoPlay` attribute, which would remove the need for this `useEffect`.